src/overlays/actors/ovl_Obj_Hgdoor/z_obj_hgdoor.c

```diff
@@ -21,36 +37,172 @@ const ActorInit Obj_Hgdoor_InitVars = {
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Hgdoor_0x80BD4090/func_80BD433C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Hgdoor_0x80BD4090/func_80BD4358.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Hgdoor_0x80BD4090/func_80BD4460.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Hgdoor_0x80BD4090/func_80BD4478.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Hgdoor_0x80BD4090/func_80BD44D0.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Hgdoor_0x80BD4090/func_80BD4500.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Hgdoor_0x80BD4090/ObjHgdoor_Update.asm")
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Hgdoor_0x80BD4090/ObjHgdoor_Draw.asm")
+s16 D_80BD4690 = 0;
+static s32 unused = 0;
+static s32 unused2 = 0;
+
+void ObjHgdoor_SetChild(ObjHgdoor* this, GlobalContext* globalCtx) {
+    Actor* actorIterator = globalCtx->actorCtx.actorList[ACTORCAT_PROP].first;
+
+    while (actorIterator) {
+        if (actorIterator->id == ACTOR_OBJ_HGDOOR && &this->dyna.actor != actorIterator) {
+            this->dyna.actor.child = actorIterator;
+            return;
+        }
+        actorIterator = actorIterator->next;
+    }
+}
+
+void ObjHgdoor_SetParent(ObjHgdoor* this, GlobalContext* globalCtx) {
+    Actor* actorIterator = globalCtx->actorCtx.actorList[ACTORCAT_PROP].first;
+
+    while (actorIterator) {
+        if (actorIterator->id == ACTOR_EN_HG) {
+            this->dyna.actor.parent = actorIterator;
+            return;
+        }
+        actorIterator = actorIterator->next;
+    }
+}
+
+void ObjHgdoor_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHgdoor* this = THIS;
+    s32 pad;
+    CollisionHeader* header = NULL;
+
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    BcCheck3_BgActorInit(&this->dyna, 1);
+    if (OBJHGDOOR_IS_RIGHT_DOOR(this)) {
+        BgCheck_RelocateMeshHeader(&D_06001D10, &header);
+    } else {
+        BgCheck_RelocateMeshHeader(&D_060018C0, &header);
+    }
+    this->dyna.bgId = BgCheck_AddActorMesh(globalCtx, &globalCtx->colCtx.dyna, &this->dyna, header);
+    this->rotation = 0;
+    this->timer = 0;
+    this->cutscene = this->dyna.actor.cutscene;
+    ObjHgdoor_SetupCheckShouldOpen(this);
+}
+
+void ObjHgdoor_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjHgdoor* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void ObjHgdoor_SetupCheckShouldOpen(ObjHgdoor* this) {
+    this->actionFunc = ObjHgdoor_CheckShouldOpen;
+}
+
+void ObjHgdoor_CheckShouldOpen(ObjHgdoor* this, GlobalContext* globalCtx) {
+    if (!(gSaveContext.weekEventReg[75] & 0x20) && !(gSaveContext.weekEventReg[52] & 0x20) &&
+        (this->dyna.actor.xzDistToPlayer < 100.0f) && (this->dyna.actor.yDistToPlayer < 40.0f) &&
+        (OBJHGDOOR_IS_RIGHT_DOOR(this))) {
```

Comment:
```suggestion
        OBJHGDOOR_IS_RIGHT_DOOR(this)) {
```
I don't think you need brackets here since there are brackets in the Macro as well